### PR TITLE
Better host deprecation information

### LINKED ISSUE #3966: Feedback: 'NServiceBus Host'
Feedback for 'NServiceBus Host' https://docs.particular.net/nservicebus/hosting/nservicebus-host/

Location in GitHub: https://github.com/Particular/docs.particular.net/blob/master/nservicebus/hosting/nservicebus-host/index.md

ping @Particular/host-maintainers 

When creating a project from a scratch targetting Host v7 and later, it doesn't update Project's debug settings to set the NServiceBus.Host.exe as a debug startup application. Would be nice if documentation could mention that. Even an entry such as "to debug in VS, update project Debug settings pointing to NServiceBus.Host.exe in the output folder. E.g.: $(solutiondir)\<project-name>\bin\Debug\<TFM>\NServiceBus.Host.exe".

### PR DESCRIPTION
**Pull request by danielmarbach**
Fixes #3966

### CODE REVIEW

File: nservicebus/hosting/nservicebus-host/host-deprecated-warning.include.md [hunk 1]
```diff
@@ -1 +1 @@
-WARNING: The NServiceBus Host is being deprecated as of version 9. Refer to [the upgrade guide](/nservicebus/upgrades/host-7to8.md) for further details.
+WARNING: For endpoints using NServiceBus version 7 it is no longer recommended to use the NServiceBus Host version 8. Use alternative approaches such as [Generic Host](/nservicebus/hosting/extensions-hosting.md), [NServiceBus Windows Service](/nservicebus/dotnet-templates.md#nservicebus-windows-service) or [NServiceBus Docker Container](/nservicebus/dotnet-templates.md#nservicebus-docker-container) instead. The host is still supported for existing endpoints to ease the transition phase to the new hosting models, refer to [the upgrade guide](/nservicebus/upgrades/host-7to8.md) for further details.
```
Comment (SeanFeldman) [inline nservicebus/hosting/nservicebus-host/host-deprecated-warning.include.md]:
```suggestion
WARNING: For endpoints using NServiceBus version 7 and above, it is no longer recommended to use the NServiceBus Host version 8. Use alternative approaches such as [Generic Host](/nservicebus/hosting/extensions-hosting.md), [NServiceBus Windows Service](/nservicebus/dotnet-templates.md#nservicebus-windows-service) or [NServiceBus Docker Container](/nservicebus/dotnet-templates.md#nservicebus-docker-container) instead. The host is still supported for existing endpoints to ease the transition phase to the new hosting models, refer to [the upgrade guide](/nservicebus/upgrades/host-7to8.md) for further details.
```

File: nservicebus/hosting/nservicebus-host/index_sample-initialization_host_[7,].partial.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+
+The following snippet shows a sample configuration that can be used as a starting point.
+
+snippet: host-endpoint-config-sample
+
+### Set the host as startup project
+
+Either setup the start project to `NServiceBus.Host.exe` (available in the output directory of the build) or create a `launchSettings.json` with the following content
```
Comment (SeanFeldman) [inline nservicebus/hosting/nservicebus-host/index_sample-initialization_host_[7,].partial.md]:
```suggestion
Either set up the start project in Visual Studio to `NServiceBus.Host.exe` (available in the output directory of the build) or create a `launchSettings.json` for VS Code with the following content
```

Given the options are different per tool used, I've spelled out which version for what tool. Feel free to discard this suggestion if you think it's unnecessary.

File: nservicebus/hosting/nservicebus-host/index_sample-initialization_host_[7,].partial.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+```json
```
Comment (SeanFeldman) [inline nservicebus/hosting/nservicebus-host/index_sample-initialization_host_[7,].partial.md]: I'd not think of including VSCode, nicely done @danielmarbach! 👍

File: nservicebus/upgrades/host-7to8.md [hunk 4]
```diff
@@ -11,19 +11,19 @@ upgradeGuideCoreVersions:
...
 ## Migrating procedure
 
-Switching to self-hosting can be done by using the [NServiceBus Windows Service](/nservicebus/dotnet-templates.md#nservicebus-windows-service) or [NServiceBus Docker Container](/nservicebus/dotnet-templates.md#nservicebus-docker-container) templates to create a new project, and then moving the relevant code and configuration over to that project.
+Switching to self-hosting can be done by using the [Generic Host](/nservicebus/hosting/extensions-hosting.md), [NServiceBus Windows Service](/nservicebus/dotnet-templates.md#nservicebus-windows-service) or [NServiceBus Docker Container](/nservicebus/dotnet-templates.md#nservicebus-docker-container) templates to create a new project, and then moving the relevant code and configuration over to that project.
 
 
 ### Configuration
 
-Self-hosting gives access to the same configuration options as provided by the host. See below for migration of host-specific configuration APIs.
+Self-hosting gives access to the same configuration options as provided by the Host. See below for migration of host-specific configuration APIs.
```
Comment (SeanFeldman) [inline nservicebus/upgrades/host-7to8.md]:
```suggestion
Self-hosting provides access to the same configuration options as provided by the Host. See below for the migration of host-specific configuration APIs.
```

File: nservicebus/upgrades/host-7to8.md [hunk 5]
```diff
@@ -44,7 +44,7 @@ The `UsingTransport<MyTransport>` role can be replaced with the equivalent `Endp
 
 #### Endpoint name
 
-The host defaults the endpoint name to the namespace of the type implementing `IConfigureThisEndpoint`. When self-hosting, that name should be passed to the constructor of an `EndpointConfiguration`.
+The Host defaults the endpoint name to the namespace of the type implementing `IConfigureThisEndpoint`. When self-hosting, that name should be passed to the constructor of an `EndpointConfiguration`.
```
Comment (SeanFeldman) [inline nservicebus/upgrades/host-7to8.md]: Other sections use the past tense. To keep it consistent, changed this one.
```suggestion
The Host defaulted the endpoint name to the namespace of the type implementing `IConfigureThisEndpoint`. When self-hosting, that name should be passed to the constructor of an `EndpointConfiguration`.
```

**Comment by SeanFeldman** [comment]
@danielmarbach should the description say `fixes ##3966`?

File: nservicebus/hosting/nservicebus-host/index_sample-initialization_host_[7,].partial.md (hunk 2 above)
Comment (danielmarbach) [reply nservicebus/hosting/nservicebus-host/index_sample-initialization_host_[7,].partial.md]: Actually launchsettings is also supported in Visual Studio. With the never version it even defaults to that.

**Comment by danielmarbach** [comment]
Thanks @SeanFeldman for the review!

**Pull request merged**